source/grid/grid_tools.cc

```diff
@@ -414,6 +415,79 @@ namespace GridTools
   }
 
 
+  template <int dim>
+  Vector<double>
+  calculate_aspect_ratio_of_cells(const Triangulation<dim> &triangulation,
+                                  const Mapping<dim> &      mapping,
+                                  const Quadrature<dim> &   quadrature)
+  {
+    FE_Nothing<dim> fe;
+    FEValues<dim>   fe_values(mapping, fe, quadrature, update_jacobians);
+
+    typedef typename Triangulation<dim>::active_cell_iterator cell_iter;
+    cell_iter cell = triangulation.begin_active(), endc = triangulation.end();
+
+    Vector<double> aspect_ratio_vector(triangulation.n_active_cells());
+
+    // loop over cells of processor
+    for (; cell != endc; ++cell)
```

Comment:
Can you change this to a range-based for loop:
```suggestion
    for (const auto &cell : triangulation.active_cell_iterators())
```
This will allow you to remove lines 427-428.

Reply:
done in https://github.com/dealii/dealii/pull/8894/commits/cd86f95e2cbfe49fee59b9e792b12e9b3921e5a6